### ISSUE TITLE
Adding withPreview() to a component wrapped in a StaticQuery causes animations to break

### ISSUE THREAD

**Issue by ClemannD** [birkir/gatsby-source-prismic-graphql#115]
I have a component that I am wrapping in a static query, and it uses the `withPreview()` helper function, which makes my previews work as expected. However, the animations no longer work on this component. I am adding/removing classes with css animations. When i remove the `withPreview()` the animations work again. Is this a limitation or is there some sort of work around?

```
class NavigationMenu extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <>
                <div className={ `nav-dropdown ${(isShowing ? '' : 'closed')}` }>
                   
                </div>
            </>
        );
    }
}

const navMenuQuery = graphql`
    {
        prismic {
            ...   
        }
    }
`;

export default (props) => (
    <StaticQuery
        query={ navMenuQuery.toString() }
        render={ withPreview((data) => (
            <NavigationMenu
                externalLinks={ data.prismic.allExternal_linkss.edges[0].node }
                socialLinks={ data.prismic.allSocialLinkss.edges[0].node }
                navMenuPreview={ data.prismic.allNavMenuPreviews.edges[0].node }
                isShowing={ props.isShowing }
            />
        ), navMenuQuery) }
    />
);
```

**Comment by codebravotech** [comment]
We had the exact same issue. I think the withPreview might (either erroneously or by design) cause re-renders on change to props of the component it's wrapping (just a theory).

The workaround for me (and the best practice in general with static queries) is to colocate every static query as close as possible to where the data is used. So rather than getting externalLinks and socialLinks both here in one big query, and then passing them as props to the component, try creating a "externalLinks" query and a "socialLinks" query for whatever component directly uses that data, and wrapping those components in StaticQuery instead.

That should fix your problem, but only if the animations are on a parent component so that you can get them "outside" of StaticQuery

**Comment by ClemannD** [comment]
Thank your for your response! I took your advice and broke down my component into smaller components each with their own queries, and the parent component now animates as expected. 

I'll leave this open for a bit incase someone is able to shed light on why it doesn't work the original way or incase it is actually a bug.

**Comment by codebravotech** [comment]
Yeah I think it definitely should be left open. My advice was just intended as a workaround. 

It's a really irritating bug, because anything that causes a state change (such as typing into an input form that uses state to hold it's value), causes a re-render which is a really ugly user experience (makes typing multiple characters at once into a form like that impossible)
